enumerable_methods.rb

```diff
@@ -0,0 +1,63 @@
+# Project 2: Enumerable Methods
+
+module Enumerable
+  def my_each(&proc)
+    length.times do |i|
+      proc.call(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index(&proc)
+    length.times do |item, _i|
+      proc.call(item)
+    end
+    self
+  end
+
+  def my_select(&proc)
+    selection = []
+    length.times do |i|
+      selection.push(self[i]) if proc.call(self[i])
+    end
+    selection
+  end
+
+  def my_all?(&proc)
+    length.times do |i|
+      return false if proc.call(self[i]) == false
+    end
+  end
+
+  def my_any?(&proc)
+    length.times do |i|
+      return true if proc.call(self[i]) == true
+    end
+  end
+
+  def my_none?(&proc)
+    length.times do |i|
+      return true if proc.call(self[i]) != true
+    end
+  end
+
+  def my_map(&proc)
+    my_each do |i|
+      self[i] = proc.call(self[i])
+    end
+    self
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_map returns an Enumerator if no block is given
   ```ruby
    array = [1,5,8,7,9,5,5,55,5]
    p array.my_map #error
    # NoMethodError:
        undefined method `call' for nil:NilClass
      # ./enumerable_methods.rb:46:in `block in my_map'
      # ./enumerable_methods.rb:6:in `block in my_each'
      # ./enumerable_methods.rb:5:in `times'
      # ./enumerable_methods.rb:5:in `my_each'
      # ./enumerable_methods.rb:45:in `my_map'
  ```

* [ ] my_map returns a new array with the results of running block once for every element in enum.
  ```ruby
    array = [1,2,5,4,7]
    block = proc { |num| num < 5 }
    p array.my_map?(&block) == array.map?(&block) # false
  ```